src/darwin/Framework/CHIP/MTRDeviceAttestationInfo.h

```diff
@@ -70,15 +70,44 @@ API_AVAILABLE(ios(16.4), macos(13.3), watchos(9.4), tvos(16.4))
  */
 @property (nonatomic, copy, readonly, nullable) NSData * firmwareInfo;
 
+/**
+ * The vendor ID value from the device's Basic Information cluster that was used
+ * for device attestation.  If attestation succeeds, this must match the vendor
+ * ID from the certification declaration.
+ */
+@property (nonatomic, readonly) NSNumber * basicInformationVendorID MTR_NEWLY_AVAILABLE;
+
+/**
+ * The product ID value from the device's Basic Information cluster that was
+ * used for device attestation.  If attestation succeeds, this must match one of
+ * the product IDs from the certification declaration.
+ */
+@property (nonatomic, readonly) NSNumber * basicInformationProductID MTR_NEWLY_AVAILABLE;
+
+/**
+ * Initializer without basicInformationVendorID and basicInformationProductID.
+ * Will set those properties to @(0).
+ */
 - (instancetype)initWithDeviceAttestationChallenge:(NSData *)challenge
                                              nonce:(NSData *)nonce
                                        elementsTLV:(MTRTLVBytes)elementsTLV
                                  elementsSignature:(NSData *)elementsSignature
                       deviceAttestationCertificate:(MTRCertificateDERBytes)deviceAttestationCertificate
          productAttestationIntermediateCertificate:(MTRCertificateDERBytes)processAttestationIntermediateCertificate
                           certificationDeclaration:(NSData *)certificationDeclaration
-                                      firmwareInfo:(NSData *)firmwareInfo;
+                                      firmwareInfo:(NSData *)firmwareInfo
+    MTR_NEWLY_DEPRECATED("Please use the version with basicInformationVendorID and basicInformationProductID");
```

Comment:
Isn't this class an output of device attestation? If so, why is the initializer even public API?

Reply:
> If so, why is the initializer even public API?

Because someone did that, we did insufficient review, an then we shipped it...  I could make the new initializer not be public, @ksperling-apple

Reply:
Yeah just deprecate the current one without a replacement and make the new one internal?